### DISCUSSION TITLE
Working with short-lived containers

### DISCUSSION THREAD

**Discussion by g0loob** [testcontainers/testcontainers-dotnet#446] [Q&A] [answered]
How to wait until container finishes its work? Or how to detect that container has exited? I have a console application in a Docker container, and I want to: 

1. start the container, 
2. wait until it's finished, 
3. do a clean up and 
4. continue with the rest of the test. 

I noticed that [Java library](https://www.testcontainers.org/features/startup_and_waits/) has `OneShotStartupCheckStrategy` and `IndefiniteWaitOneShotStartupCheckStrategy` for such scenarios, but I can't find anything similar for .NET.

**Comment by HofmeisterAn** [comment, marked as answer]
You can use a custom wait strategy. Similar to:

https://github.com/HofmeisterAn/dotnet-testcontainers/blob/de4c4a251152ee2cfdcd4061c8b1bcc0621e692f/src/DotNet.Testcontainers/Configurations/WaitStrategies/UntilContainerIsRunning.cs#L14

either wait until the container reaches the exit state or use whatever signals a finished run. `StartAsync` completes when the wait strategies are successful.

`GetExitCode` returns the exit code of the container.
